src/index.d.ts

```diff
@@ -0,0 +1,693 @@
+import { Script, TxOut } from 'bsv';
+
+declare interface CastTemplate {
+  lockingScript: {
+    script: unknown[];
+    size: ((params: object) => number) | number;
+    validate: (params: object) => void;
+  }
+}
```

Comment:
This is a tricky definition to start with, but the cast template ordinarily contains two properties: `lockingScript` and `unlockingScript`. It's not necessary to have both - in some cases only a lockingScript or only and unlockingScript will exist in the Cast. But at least one is necessary.

Both locking and unlocking script are objects with the following definition:

* script - required - array of elements which can either be a number, a buffer, or a function (which returns a number of buffer)
* size - required - number or function (you got that right)
* setup - optional - a function that receives the params object and returns an object that is merged in to the params
* validate - optional - a function that receives the params object and throws an error if theres a problem with the params 

And another gotcha. Whilst both lockingScript and unlockingScript templates are the same, the functions in an unlockingScript receive an additional 1st parameter - a ctx object - the params object is the second parameter

Hope that makes sense 🥴 ask any questions if not

Reply:
Sorted, I think! Apologies for the delay... :eyes: